### PR TITLE
add changeset and changeset ci workflow

### PR DESCRIPTION
**Pull request by momentmaker**
This adds `changesets` as the default way to update the CHANGELOG.

- adds `pnpm changesets` with instruction
- sets up changesets with github-changelog
- adds changesets checker gha workflow to check if any changes in `core`, `common`, `plugins`
- updates the `.gitignore`

### CODE REVIEW

File: .changeset/config.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@2.3.1/schema.json",
+  "changelog": [
+    "@changesets/changelog-github",
```
Comment (chainchad) [inline .changeset/config.json:4]: When adding this, does it require all eng's to have a local GH token set in order to run a `pnpm changeset` command to create the changeset markdwon? Or, is this only needed when running other subcommands like `pnpm changeset version`?

**Review by chainchad — APPROVED** [review]

File: .changeset/config.json (hunk 1 above)
Comment (momentmaker) [reply .changeset/config.json:4]: oh interesting... I didn't even realize it needs a gh_token.

it seems to work fine when other devs are adding a new changeset in core when running `pnpm changeset`

I ran `pnpm changeset version` too and I don't think it was using the gh_token.

Comment (momentmaker) [reply .changeset/config.json:4]: I think only `publish` needs the gh_token?

https://dev.to/ignace/automate-npm-releases-on-github-using-changesets-25b8

Comment (chainchad) [reply .changeset/config.json:4]: Ah, it must have just been on publish 👍

**Comment by momentmaker** [comment]
@RensR @connorwstein @jasonmci hey guys, could you please review this?

**Review by AnieeG — APPROVED** [review]

**Pull request merged**